### PR TITLE
Parameter fixes

### PR DESCRIPTION
**Pull request by K20shores**
Fixed a few logic issues in the solver

- calculating the jacobian exactly once for each phase of the solver
- calculating an initial forcing which I had removed previously, not realizing that when rejecting an `H`, the initial forcing is used
- setting some of the same parameters as KPP
- setting the right size for `Yerror`
- added a `NaN` check

### CODE REVIEW

File: include/micm/solver/rosenbrock.hpp [hunk 1]
```diff
@@ -76,7 +76,7 @@ namespace micm
     // Gamma_i = \sum_j  gamma_{i,j}
     std::array<double, 6> gamma_{};
 
-    double absolute_tolerance_{ 1e-12 };
+    double absolute_tolerance_{ 1e-3 };
```
Comment (mattldawson) [inline include/micm/solver/rosenbrock.hpp:79]: I'm going to create a separate issue to figure out how to handle absolute tolerances. They will depend on the units, which for just the gas-phase we can maybe base them on our default units of `mol m-3`, but if we add aerosol-phase species, we might have to rethink this.

File: include/micm/solver/rosenbrock.hpp [hunk 2]
```diff
@@ -585,14 +587,14 @@ namespace micm
 
    protected:
     /// @brief Computes the scaled norm of the vector errors
-    /// @param original_number_densities the original number densities
-    /// @param new_number_densities the new number densities
+    /// @param Y the original vector
+    /// @param new_number_densities the new vector
```
Comment (mattldawson) [inline include/micm/solver/rosenbrock.hpp:591]:
```suggestion
    /// @param Ynew the new vector
```

File: include/micm/solver/rosenbrock.hpp [hunk 3]
```diff
@@ -585,14 +587,14 @@ namespace micm
...
     /// @param errors The computed errors
     /// @return
     double NormalizedError(
-        MatrixPolicy<double> original_number_densities,
-        MatrixPolicy<double> new_number_densities,
-        MatrixPolicy<double> errors);
+      const MatrixPolicy<double>& Y,
+      const MatrixPolicy<double>& Ynew,
```
Comment (mattldawson) [inline include/micm/solver/rosenbrock.hpp:596]: if we're changing the names, we should probably follow our style of:
```suggestion
      const MatrixPolicy<double>& y,
      const MatrixPolicy<double>& y_new,
```

Comment (mattldawson) [reply include/micm/solver/rosenbrock.hpp:596]: or we can do this with #129

**Review by mattldawson — APPROVED** [review]
looks great! just a couple very minor comments

**Review by boulderdaze — APPROVED** [review]

**Pull request merged**